File: backend/WebApi/Lunchorder.Domain/Dtos/MenuEntry.cs
using System;

namespace Lunchorder.Domain.Dtos
{
    /// <summary>
    /// A menu entry
    /// </summary>
    public class MenuEntry
    {
        private string _picture;

        /// <summary>
        /// An identifier for the menu entry
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// The name of the menu entry
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// The description of the menu entry
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// The associated category for the menu entry
        /// </summary>
        public string CategoryId { get; set; }

        /// <summary>
        /// An URL that contains a picture for the menu entry
        /// </summary>
        public string Picture
        {
            get { return string.IsNullOrEmpty(_picture) ? "" : _picture; }
            set { _picture = value; }
        }

        /// <summary>
        /// The price for the menu entry
        /// </summary>
        public string Price { get; set; }

        /// <summary>
        /// Represents if the entry is in an enabled state
        /// </summary>
        public bool Enabled { get; set; }
    }
}

File: frontend/src/app/domain/dto/menuEntry.ts
import { MenuRule } from './menuRule'

export class MenuEntry implements app.domain.dto.IMenuEntry, Serializable<MenuEntry> {
    id: string;
    name: string;
    description: string;
    categoryId: string;
    picture: string;
    price: number;
    enabled: boolean;
    rules: MenuRule[];

    deserialize(input: any): MenuEntry {
        this.id = input.id;
        this.name = input.name;
        this.description = input.description;
        this.categoryId = input.categoryId;
        this.picture = input.picture;
        this.price = input.price;
        this.enabled = input.enabled;

        this.rules = new Array<MenuRule>();
        if (input.rules) {
            for (var rule of input.rules) {
                this.rules.push(new MenuRule().deserialize(rule));
            }
        }
        return this;
    }
}